### PR TITLE
Bug 1618578 - All perfherder dropdowns have the items in alphabetical order

### PR DESCRIPTION
**Pull request by kimberlythegeek**
removed the sorting from dropdown menu items

### CODE REVIEW

**Comment by kimberlythegeek** [comment]
The pinned projects were not being displayed in the correct order, but it was a simple fix. Should I include that on this PR?

**Comment by airimovici** [comment]
I think it can be included here

File: ui/shared/DropdownMenuItems.jsx [hunk 1]
```diff
@@ -33,14 +33,12 @@ const DropdownMenuItems = ({
     {pinned.length > 0 &&
       options
```
Comment (airimovici) [inline ui/shared/DropdownMenuItems.jsx]: As you said above, this doesn't show the pinned items in the defined order. The fix can be included in this PR

**Comment by kimberlythegeek** [comment]
@sarah-clements r?

File: ui/perfherder/graphs/TestDataModal.jsx [hunk 2]
```diff
@@ -429,6 +429,7 @@ export default class TestDataModal extends React.Component {
 
     return (
       <Modal size="lg" isOpen={showModal}>
+        {this.state.pinnedProjects}
```
Comment (sarah-clements) [inline ui/perfherder/graphs/TestDataModal.jsx]: You can use destructuring assignment for `pinnedProjects`.

Comment (kimberlythegeek) [reply ui/perfherder/graphs/TestDataModal.jsx]: ah, thanks for catching that. I had already included it in the destructuring assignment above but was still referencing `this.state` directly

File: ui/shared/DropdownMenuItems.jsx [hunk 3]
```diff
@@ -31,16 +31,14 @@ const DropdownMenuItems = ({
   <DropdownMenu className="overflow-auto dropdown-menu-height">
     {/* Items pinned to top of dropdown */}
     {pinned.length > 0 &&
-      options
-        .filter(item => pinned.includes(item))
-        .sort((a, b) => a > b)
+      pinned
+        .filter(item => options.includes(item))
```
Comment (ionutgoldan) [inline ui/shared/DropdownMenuItems.jsx]: We've already performed this filtering in the client code, so this step is redundant.

**Comment by ionutgoldan** [comment]
@kimberlythegeek please rebase this PR on a daily basis, until it gets merged. Thanks!

File: ui/perfherder/graphs/TestDataModal.jsx [hunk 4]
```diff
@@ -429,6 +429,7 @@ export default class TestDataModal extends React.Component {
 
     return (
       <Modal size="lg" isOpen={showModal}>
+        {pinnedProjects}
```
[the remark this answers is not in the document]
Comment (kimberlythegeek) [reply ui/perfherder/graphs/TestDataModal.jsx]: Thought I removed this already

**Comment by kimberlythegeek** [comment]
creating a new PR to clean up history 🙃

**Comment by sarah-clements** [comment]
> creating a new PR to clean up history 🙃

FYI, the best way to keep a clean history is to rebase your branch onto upstream/master (set the upstream remote to point at the mozilla/treeherder repo) instead of merging from your fork. You can also use rebase to drop a bunch of commits you didn't need to pull in.

**Comment by kimberlythegeek** [comment]
Thanks @sarah-clements , rebasing isn't something I've had to do often. sometimes I get it right and sometimes I don't, haha. Alex Ionescu shared a git reference that should be helpful in the future

**Pull request closed without merging**